keep/api/bl/enrichments_bl.py

```diff
@@ -343,7 +343,7 @@ def _check_alert_matches_rule(self, alert: AlertDto, rule: MappingRule) -> bool:
     def _is_match(value, pattern):
         if value is None or pattern is None:
             return False
-        return re.match(pattern, value) is not None
+        return re.search(pattern, value) is not None
```

Comment:
Or should it be `re.fullmatch` ? What is the intentional usage?

Reply:
Asked GPT to summarize the difference for us :)

	•	re.match(): Checks for a match only at the start of the string.
	•	re.search(): Searches for a match anywhere in the string.
	•	re.fullmatch(): Ensures the pattern matches the entire string.

I think `re.search` is our winner as we want to search for that pattern anywhere in the string.

Reply:
I think it should be `fullmatch` as it otherwise could cause problems with mapping and would maybe defeat the propose of extraction?

Reply:
I'm not sure @mfroembgen because I think fullmatch expects the patterns to match the *whole* string.
In some example, where somebody wants to extract severity and name from an alert message:
`INFO: The name of the alert - <TIMESTAMP> - <SOMETHING ELSE>`
it might not work, wdyt?

Reply:
@talboren For extraction I totally agree but should it behave the same for the mapping as well. Because I would have thought to extract for example the severity from the alert and utilise the extracted severity then for the mapping. Lets say to add P1-5 if severity is INFO and so on.

Reply:
I think it makes sense for mapping as well - your use case would still work.